In [1]:
import json
import pandas as pd
import xlrd
import os

# Salto

In [2]:
with open('../data/data_salto/concursos/concursos_salto_nacional_2025.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Nombre es: 124
La longitud de Categoría es: 124
La longitud de Provincia es: 124
La longitud de Localidad es: 124
La longitud de Disciplina es: 124
La longitud de Federación es: 124
La longitud de Resultados es: 124
La longitud de País es: 124
La longitud de Inicio es: 124
La longitud de Final es: 124
La longitud de Ambito es: 124


In [2]:
def creacion_columnas(df, columna_crear, columna, coso = "paises"):
    posiciones = ["RET", "ELI", "NOP"]

    df[columna] = df[columna].astype(str)
    
    if coso == "paises":
        df[columna_crear] = df[columna].apply(lambda x: x.split("(")[1].strip(")") if x.endswith(")") else "ESP")
        df[columna] = df[columna].apply(lambda x: x.split("(")[0].strip() if x.endswith(")") else x)
    elif coso == "Estado": 
        if "Puntuacion" in df.columns:  
            df["Posicion"] = df.apply(lambda x: "NOP" if pd.isna(x["Puntuacion"]) else x["Posicion"], axis=1)
        df[columna_crear] = df[columna].apply(lambda x: x if x in posiciones else "FIN")
    elif coso == "Premio":
        df[columna_crear] = df[columna].apply(lambda x: False if x == 0 else True)
        
def modificaciones_generales(dataframe):

    dataframe["Prueba"] = dataframe.iloc[3,1]
    dataframe["Fecha_prueba"] = dataframe.iloc[2,1]
    dataframe["Concurso"] = dataframe.iloc[1,1]
    dataframe = dataframe.iloc[9:].reset_index(drop = True)
    dataframe = dataframe.rename(columns = {"Unnamed: 0": "Posicion", 
                                                 "Unnamed: 1": "Lic_jinete",
                                                 "Unnamed: 2": "Jinete", 
                                                 "Unnamed: 3": "Lic_caballo", 
                                                 "Unnamed: 4": "Caballo",
                                                 "Unnamed: 5": "Raza_caballo",
                                                 "Unnamed: 6": "Puntuacion",
                                                 "Unnamed: 7": "Dinero_premio"})
    dataframe["Raza_caballo"] = dataframe["Raza_caballo"].apply(lambda x: None if x == "--" else x)
    creacion_columnas(dataframe, "Pais_jinete", "Jinete")
    creacion_columnas(dataframe, "Pais_caballo", "Caballo")
    creacion_columnas(dataframe, "Estado", "Posicion", coso = "Estado")
    creacion_columnas(dataframe, "Premio", "Dinero_premio", coso = "Premio")
    dataframe = dataframe.reindex(columns = ['Estado','Posicion', 'Lic_jinete', 'Jinete', 'Pais_jinete', 'Lic_caballo', 'Caballo',
              'Raza_caballo', 'Pais_caballo', 'Puntuacion', 'Premio','Dinero_premio', 'Prueba', 'Fecha_prueba',
              'Concurso'])

    return dataframe

In [3]:
directorio = "../data/data_salto/resultados_2025"

concursos = []
tipos_pruebas = []
combinaciones_resultados = []
# Obtener la lista de todos los archivos en el directorio
archivos = os.listdir(directorio)

# Filtrar solo los archivos que parecen ser de Excel (puedes refinar esto más si es necesario)
archivos_excel = [f for f in archivos if f.endswith(('.xlsx', '.xls'))]

if archivos_excel:
    # Ordenar la lista de archivos para asegurar que el "primero" sea consistente
    archivos_excel.sort()

    for nombre_excel in archivos_excel:
        ruta_excel = os.path.join(directorio, nombre_excel)
        print(f"\n--- Leyendo archivo: {nombre_excel} ---")
        try:
            # Leer el archivo Excel actual
            df = pd.read_excel(ruta_excel)
            concursos.append(df.iloc[1,1]) # Mostrar las primeras filas de cada DataFrame
            tipos_pruebas.append(df.iloc[3,1])
            df = modificaciones_generales(df)
            combinaciones_resultados.append(df["Puntuacion"])
        except FileNotFoundError:
            print(f"Error: No se encontró el archivo: {ruta_excel}")
        except Exception as e:
            print(f"Ocurrió un error al leer el archivo {nombre_excel}: {e}")
else:
        print("No se encontraron archivos Excel en el directorio.")            


--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (1).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (10).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (100).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (11).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (12).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (13).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (14).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (15).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (16).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (17).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (18).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (19).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (2).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (20).xls ---

--- Leyendo archivo:

In [4]:
# comprobamos que el numero de excels es correcto
print(len(concursos))
print(len(tipos_pruebas))
print(len(combinaciones_resultados))

1518
1518
1518


In [5]:
combinaciones_resultados

[0                      0/59,91
 1                      0/63,23
 2                      0/65,58
 3                      0/66,00
 4                      0/66,78
 5                      0/67,65
 6                      0/68,85
 7                      0/70,17
 8                      0/71,27
 9                      0/72,72
 10                     0/73,55
 11                     0/74,63
 12                     0/75,85
 13                     0/76,22
 14                     0/77,36
 15                     0/77,68
 16                     0/77,90
 17      4(4 Obs - 0 Tpo)/67,67
 18      4(4 Obs - 0 Tpo)/71,10
 19      4(4 Obs - 0 Tpo)/71,48
 20      4(4 Obs - 0 Tpo)/72,98
 21      4(4 Obs - 0 Tpo)/73,04
 22      4(4 Obs - 0 Tpo)/73,27
 23      4(4 Obs - 0 Tpo)/73,31
 24      4(4 Obs - 0 Tpo)/74,34
 25      4(4 Obs - 0 Tpo)/74,62
 26      4(4 Obs - 0 Tpo)/76,25
 27      4(4 Obs - 0 Tpo)/77,77
 28      4(4 Obs - 0 Tpo)/79,90
 29      5(0 Obs - 5 Tpo)/85,10
 30      8(8 Obs - 0 Tpo)/67,65
 31     

In [5]:
# obtenemos los valores unicos de la pruebas
valores_unicos = list(set(tipos_pruebas))
print(len(valores_unicos))
valores_unicos


170


['1,20 - Dos fases Especial (Art. 274.2) (274.2)',
 '1,45 (GP) - A.c.c. y desempate al cronómetro (238.2.2)',
 '1,30 - Caza (239)',
 'Ch - 1,25 m. - 2 fases Asc / Acc (274.1.5.2)',
 '1,10 - Dos Fases (274.1.5.3)',
 '1,10 - A con cronómetro (238.2.1)',
 'YR - 1,40 m. - A c.c. (238.2.1)',
 '1,40 (GP) - A.c.c. y desempate (238.2.2)',
 '1,30 - Accumulator (269.3.2)',
 '6 años - 1,25 - A c.c. (238.2.1)',
 '7 años - 1,35 - A.c.c. y desempate (238.2.2)',
 '6 años - 1,20 - A c.c. (238.2.1)',
 '1,40 (M) - A c.c. (238.2.1)',
 'YR - 1,45 m. - A c.c. (238.2.1)',
 '1,40 gp - Dos Mangas sin desempate (273.3.3.1)',
 '1.15 - Acc (238.2.1)',
 '1,30 - A c.c. (238.2.1)',
 '1,45 - A c.c. (238.2.1)',
 '8 años - 1,40 - A.c.c. (238.2.1)',
 '1,20 - Dos Fases Especial (274.2)',
 'J - CN - C.N. (264)',
 '1,50 (G) - A con cronometro sin desempate (238.2.1)',
 '5 años - 1,10 - A c.c. (238.2.1)',
 '1,30 - Dificultades Progresivas (269.3.2)',
 '1,35 GP - Dos Mangas (A c/c - A c/c) (273.3.3.1)',
 '7 años - 1,40 - A 

In [6]:
# como queremos quedarnos con la parte general de las pruebas, lo sacamos de cada tipo de prueba
tipos_pruebas_bueno = []
for elemento in valores_unicos:
    elemento_bueno = elemento.split("-")
    if len(elemento_bueno) == 3:
        tipos_pruebas_bueno.append(elemento_bueno[2].lstrip())
    elif len(elemento_bueno) == 2:
        tipos_pruebas_bueno.append(elemento_bueno[1].lstrip())

tipos_pruebas_bueno

['Dos fases Especial (Art. 274.2) (274.2)',
 'A.c.c. y desempate al cronómetro (238.2.2)',
 'Caza (239)',
 '2 fases Asc / Acc (274.1.5.2)',
 'Dos Fases (274.1.5.3)',
 'A con cronómetro (238.2.1)',
 'A c.c. (238.2.1)',
 'A.c.c. y desempate (238.2.2)',
 'Accumulator (269.3.2)',
 'A c.c. (238.2.1)',
 'A.c.c. y desempate (238.2.2)',
 'A c.c. (238.2.1)',
 'A c.c. (238.2.1)',
 'A c.c. (238.2.1)',
 'Dos Mangas sin desempate (273.3.3.1)',
 'Acc (238.2.1)',
 'A c.c. (238.2.1)',
 'A c.c. (238.2.1)',
 'A.c.c. (238.2.1)',
 'Dos Fases Especial (274.2)',
 'C.N. (264)',
 'A con cronometro sin desempate (238.2.1)',
 'A c.c. (238.2.1)',
 'Dificultades Progresivas (269.3.2)',
 'A c/c) (273.3.3.1)',
 'A c.c. (238.2.1)',
 'A.c.c. y desempate (238.2.2)',
 'Acc (238.2.1)',
 'A c.c. (238.2.1)',
 'A c.c. (238.2.1)',
 '2 fases Asc / Acc (274.1.5.2)',
 'A.c.c. y desempate (238.2.2)',
 'Dos Fases Especial (274.2)',
 'Dos Fases Especial (274.2)',
 'Dos Fases (274.1.5.3)',
 'A.c.c. y desempate (238.2.2)',
 'A.c.c.

In [7]:
# volvemos a obtener los valores unicos de los tipos de pruebas
pruebas_unicas = list(set(tipos_pruebas_bueno))
pruebas_unicas

['A.c.c. (238.2.1)',
 'A.c.c. y desempate al cronómetro (238.2.2)',
 'A.c.c.) sin desempate (273.3.3.1)',
 'Dos Fases (274.1.5.3)',
 'A c/c y un desempate (238.2.2)',
 'Acc con desempate (238.2.2)',
 'A.c.c y desempate (238.2.2)',
 'Accumulator (269.3.2)',
 'Acc (238.2.1)',
 'A.c.c (238.2.1)',
 'Dificultades Progresivas con Jocker (269.3.2)',
 'Dificultades Progresivas (269.3.2)',
 'A.s.c. y desempate al cronómetro (238.1.2)',
 'Velocidad y Manejabilidad (263)',
 'Dos fases Especial (Art. 274.2) (274.2)',
 'DOS FASES (274.1.5.3)',
 'Dos Fases Especial (274.2)',
 'A.c.c. y desempate (238.2.2)',
 'Acc y desempate (238.2.1)',
 'C (239)',
 'Dos Mangas sin desempate (273.3.3.1)',
 '2 fases Asc / Acc (274.1.5.2)',
 '2 FASES (274.1.5.2)',
 'Caza (239)',
 'A sin cronómetro (238.1.1)',
 'Manga Ganadora (276.2)',
 'C.N. (264)',
 'Dos Mangas (273.3.4)',
 'A con cronómetro (238.2.1)',
 'A c.c. (238.2.1)',
 'Acc y desempate (238.2.2)',
 'A c/c) (273.3.3.1)',
 'Acc (238.1.1)',
 'A c/c) (274.1.5.3)',

In [8]:
# comprobamos cuantos tipos de pruebas unicas tenemos
len(pruebas_unicas)

35

In [9]:
normas = []
for elemento in pruebas_unicas:
    tipo_norma = elemento.split(" ")
    normas.append(tipo_norma[-1])

normas

['(238.2.1)',
 '(238.2.2)',
 '(273.3.3.1)',
 '(274.1.5.3)',
 '(238.2.2)',
 '(238.2.2)',
 '(238.2.2)',
 '(269.3.2)',
 '(238.2.1)',
 '(238.2.1)',
 '(269.3.2)',
 '(269.3.2)',
 '(238.1.2)',
 '(263)',
 '(274.2)',
 '(274.1.5.3)',
 '(274.2)',
 '(238.2.2)',
 '(238.2.1)',
 '(239)',
 '(273.3.3.1)',
 '(274.1.5.2)',
 '(274.1.5.2)',
 '(239)',
 '(238.1.1)',
 '(276.2)',
 '(264)',
 '(273.3.4)',
 '(238.2.1)',
 '(238.2.1)',
 '(238.2.2)',
 '(273.3.3.1)',
 '(238.1.1)',
 '(274.1.5.3)',
 '(238.2.1)']

In [10]:
len(normas)

35

In [11]:
# volvemos a obtener los valores unicos de los tipos de pruebas
normas_unicas = list(set(normas))
normas_unicas

['(239)',
 '(238.2.1)',
 '(238.2.2)',
 '(269.3.2)',
 '(263)',
 '(274.1.5.2)',
 '(264)',
 '(274.2)',
 '(273.3.3.1)',
 '(238.1.1)',
 '(276.2)',
 '(238.1.2)',
 '(274.1.5.3)',
 '(273.3.4)']

In [12]:
len(normas_unicas)

14

In [13]:
for prueba in pruebas_unicas:
    norma = normas_unicas[0]
    if isinstance(norma, str) and norma in prueba:
        print(prueba)
    else:
        continue

C (239)
Caza (239)


In [ ]:
def limpieza_df(df):

    df = modificaciones_generales(df)
    df["Puntuacion"] = df["Puntuacion"].str.split("\n")
    celda = df.iloc[0, -6]

    if len(celda) == 1 and "/" in celda[0]:
        df["Ptos_Obs_R1"] = df["Puntuacion"].apply(lambda x: x[0].split("/")[0])
        df["Ptos_tiempo_R1"] = None
        df["Tiempo_R1"] = None

        df["Ptos_Obs_R2"] = None
        df["Ptos_tiempo_R2"] = None
        df["Tiempo_R2"] = None

        df["Ptos_Obs_R3"] = None
        df["Ptos_tiempo_R3"] = None
        df["Tiempo_R3"] = None
        
    elif len(celda) == 1 and "/" not in celda[0]:
        
        df["Ptos_Obs_R1"] = df["Puntuacion"].apply(lambda x: x[0])
        df["Ptos_tiempo_R1"] = None
        df["Tiempo_R1"] = None

        df["Ptos_Obs_R2"] = None
        df["Ptos_tiempo_R2"] = None
        df["Tiempo_R2"] = None

        df["Ptos_Obs_R3"] = None
        df["Ptos_tiempo_R3"] = None
        df["Tiempo_R3"] = None
    

    # try: 
    #     df[["Puntuacion_1", "Puntuacion_2", "Puntuacion_3"]] = df["Puntuacion"].str.split("\n", expand = True)

    # except ValueError:

    #     try:
    #         df[["Puntuacion_1", "Puntuacion_2"]] = df["Puntuacion"].str.split("\n", expand = True)
    #         df["Puntuacion_3"] = None

    #     except ValueError:

    #         if len 
    #         df["Ptos_Obs_R1"] = df["Puntuacion"].str.split("/")
    #         df["Ptos_Obs_R2"]
    #         df["Ptos_Obs_R3"]
    #         df["Puntuacion_1"] = df["Puntuacion"]
    #         df["Puntuacion_2"] = None
    #         df["Puntuacion_3"] = None

    df = df.drop("Puntuacion", axis = 1)
    return df

In [114]:
coso = "4(4 Obs - 0 Tpo)/70,88"
coso.split("/")

coso

'4(4 Obs - 0 Tpo)/70,88'

In [34]:
normas_unicas

['(239)',
 '(238.2.1)',
 '(238.2.2)',
 '(269.3.2)',
 '(263)',
 '(274.1.5.2)',
 '(264)',
 '(274.2)',
 '(273.3.3.1)',
 '(238.1.1)',
 '(276.2)',
 '(238.1.2)',
 '(274.1.5.3)',
 '(273.3.4)']

In [8]:
# comprobamos los excels que son de tipo Acc con desempate
directorio = "../data/data_salto/resultados_2025"
concursos = []
tipos_pruebas = []
combinaciones_resultados = []
# Obtener la lista de todos los archivos en el directorio
archivos = os.listdir(directorio)

# Filtrar solo los archivos que parecen ser de Excel (puedes refinar esto más si es necesario)
archivos_excel = [f for f in archivos if isinstance(f, str) and f.endswith(('.xls', '.xlsx'))]

if archivos_excel:
    # Ordenar la lista de archivos para asegurar que el "primero" sea consistente
    archivos_excel.sort()

    for nombre_excel in archivos_excel:
        ruta_excel = os.path.join(directorio, nombre_excel)
        print(f"\n--- Leyendo archivo: {nombre_excel} ---")
        try:
            # Leer el archivo Excel actual
            df = pd.read_excel(ruta_excel)
            concursos.append(df.iloc[1,1]) # Mostrar las primeras filas de cada DataFrame
            tipos_pruebas.append(df.iloc[3,1])
            df = modificaciones_generales(df)
            combinaciones_resultados.append(df["Puntuacion"].tolist())
        except FileNotFoundError:
            print(f"Error: No se encontró el archivo: {ruta_excel}")
        except Exception as e:
            print(f"Ocurrió un error al leer el archivo {nombre_excel}: {e}")
else:
        print("No se encontraron archivos Excel en el directorio.") 

# if dfs:
#     df_concat = pd.concat(dfs, ignore_index = True)
#     print("df concatenado")
# else:
#     print("No se encontraron DataFrames para concatenar.")
    
# print(norma_buscar)  



--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (1).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (10).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (100).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (11).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (12).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (13).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (14).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (15).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (16).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (17).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (18).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (19).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (2).xls ---

--- Leyendo archivo: RFHE_SALTO_Resultados de la Prueba (20).xls ---

--- Leyendo archivo:

In [11]:
len(combinaciones_resultados)

1479

## Pruebas norma (264): Copa de naciones

Son a dos rondas siempre, a excepción de que haya empate entre los primeros equipos, caso en el que habrá un desempate, llevando así a haber tres rondas

In [168]:
import numpy as np

In [219]:
def separacion_puntuaciones(dataframe, puntuacion,):
    if "/" in puntuacion_r1:
    partes = puntuacion_r1.split("/")

    if "(" not in partes[0]:
        cosita.at[indice, "Ptos_Obs_R1"] = partes[0]
        cosita.at[indice, "Tiempo_R1"] = partes[1]
    elif "(" in partes[0]:
        ptos_obs = partes[0].split("(")[1].split("-")[0][:-5].strip()
        ptos_tiempo = partes[0].split("(")[1].split("-")[1][:-5].strip()
        cosita.at[indice, "Ptos_Obs_R1"] = ptos_obs
        cosita.at[indice, "Ptos_tiempo_R1"] = ptos_tiempo
        cosita.at[indice, "Tiempo_R1"] = partes[1]
    else:
                cosita.at[indice, "Ptos_Obs_R1"] = puntuacion_r1

SyntaxError: incomplete input (1931059392.py, line 1)

In [16]:

cosita = pd.read_excel("../data/data_salto/resultados_2025/RFHE_SALTO_Resultados de la Prueba - 2025-05-13T195410.980.xls")
cosita.head(10)

Unnamed: 0                                         Unnamed: 1  \
0         NaN                                                NaN   
1   Concurso:  SPRING MET IV - CSI3* GOLD (24/04/2025 - 27/04...   
2      Fecha:                                2025-04-26 00:00:00   
3     Prueba:                       1,30 - Dos Fases (274.1.5.3)   
4  Categoría:                                             CSI***   
5         NaN                                                NaN   
6         NaN                                                NaN   
7      Puesto                                     Jinete/Amazona   
8         NaN                                                FEI   
9           1                                           10075553   

              Unnamed: 2 Unnamed: 3         Unnamed: 4 Unnamed: 5  \
0                    NaN        NaN                NaN        NaN   
1                    NaN        NaN                NaN        NaN   
2                    NaN        NaN                NaN        NaN   
3                    NaN        NaN                NaN        NaN   
4                    NaN        NaN                NaN        NaN   
5                    NaN        NaN                NaN        NaN   
6                    NaN        NaN                NaN        NaN   
7                    NaN    Caballo                NaN        NaN   
8                 Nombre        FEI             Nombre       Raza   
9  Michael G DUFFY (IRL)    108OF75  CALANDRO 21 (GER)         --   

         Unnamed: 6 Unnamed: 7  
0               NaN        NaN  
1               NaN        NaN  
2               NaN        NaN  
3               NaN        NaN  
4               NaN        NaN  
5               NaN        NaN  
6               NaN        NaN  
7        Puntuación     Premio  
8               NaN        NaN  
9  0/48,65\n0/26,98        250

In [17]:
cosita = pd.read_excel("../data/data_salto/resultados_2025/RFHE_SALTO_Resultados de la Prueba - 2025-05-13T173909.958.xls")
cosita.head(10)
# cosita["Puntuacion"] = cosita["Puntuacion"].str.split("\n")
# cosita
# for indice, row in cosita.iterrows():
#     celda = row["Puntuacion"]
#     if pd.isnull(celda):
#         continue
#     else:
#         celda = celda.split("\n")
    
#         if len(celda) == 3:
#             puntuacion_r1 = celda[0]  # Accedemos al primer elemento de la lista, que en este caso tiene solo un elemento, el 0
#             puntuacion_r2 = celda[1]
#             puntuacion_r3 = celda[2]
#             if "/" in puntuacion_r1:
#                 partes = puntuacion_r1.split("/")

#                 if "(" not in partes[0]:
#                     cosita.at[indice, "Ptos_Obs_R1"] = partes[0]
#                     cosita.at[indice, "Tiempo_R1"] = partes[1]
#                 elif "(" in partes[0]:
#                     ptos_obs = partes[0].split("(")[1].split("-")[0][:-5].strip()
#                     ptos_tiempo = partes[0].split("(")[1].split("-")[1][:-5].strip()
#                     cosita.at[indice, "Ptos_Obs_R1"] = ptos_obs
#                     cosita.at[indice, "Ptos_tiempo_R1"] = ptos_tiempo
#                     cosita.at[indice, "Tiempo_R1"] = partes[1]
#             else:
#                 cosita.at[indice, "Ptos_Obs_R1"] = puntuacion_r1
            
#             if "/" in puntuacion_r2:
#                 partes = puntuacion_r2.split("/")

#                 if "(" not in partes[0]:
#                     cosita.at[indice, "Ptos_Obs_R2"] = partes[0]
#                     cosita.at[indice, "Tiempo_R2"] = partes[1]
#                 elif "(" in partes[0]:
#                     ptos_obs = partes[0].split("(")[1].split("-")[0][:-5].strip()
#                     ptos_tiempo = partes[0].split("(")[1].split("-")[1][:-5].strip()
#                     cosita.at[indice, "Ptos_Obs_R2"] = ptos_obs
#                     cosita.at[indice, "Ptos_tiempo_R2"] = ptos_tiempo
#                     cosita.at[indice, "Tiempo_R2"] = partes[1]
#             else:
#                 cosita.at[indice, "Ptos_Obs_R2"] = puntuacion_r2

#             if "/" in puntuacion_r3:
#                 partes = puntuacion_r3.split("/")

#                 if "(" not in partes[0]:
#                     cosita.at[indice, "Ptos_Obs_R3"] = partes[0]
#                     cosita.at[indice, "Tiempo_R3"] = partes[1]
#                 elif "(" in partes[0]:
#                     ptos_obs = partes[0].split("(")[1].split("-")[0][:-5].strip()
#                     ptos_tiempo = partes[0].split("(")[1].split("-")[1][:-5].strip()
#                     cosita.at[indice, "Ptos_Obs_R3"] = ptos_obs
#                     cosita.at[indice, "Ptos_tiempo_R3"] = ptos_tiempo
#                     cosita.at[indice, "Tiempo_R3"] = partes[1]
#             else:
#                 cosita.at[indice, "Ptos_Obs_R3"] = puntuacion_r3

# cosita

# cosita
# try:
#     cosita[["Puntuacion_1", "Puntuacion_2", "Puntuacion_3"]] = cosita["Puntuacion"].str.split("\n", expand = True)

# except ValueError:
#     cosita[["Puntuacion_1", "Puntuacion_2"]] = cosita["Puntuacion"].str.split("\n", expand = True)
#     cosita["Puntuacion_3"] = None
# cosita = cosita.drop("Puntuacion", axis = 1)
# cosita.head(5)

Unnamed: 0                                         Unnamed: 1  \
0         NaN                                                NaN   
1   Concurso:  Andalucía Sunshine Tour CSIOYJCh 2025 (13/03/2...   
2      Fecha:                                2025-03-14 00:00:00   
3     Prueba:                                J - CN - C.N. (264)   
4  Categoría:                                        CSIO Ch/J/Y   
5         NaN                                                NaN   
6         NaN                                                NaN   
7      Puesto                                     Jinete/Amazona   
8         NaN                                                FEI   
9           1                                           10188567   

           Unnamed: 2 Unnamed: 3   Unnamed: 4 Unnamed: 5  \
0                 NaN        NaN          NaN        NaN   
1                 NaN        NaN          NaN        NaN   
2                 NaN        NaN          NaN        NaN   
3                 NaN        NaN          NaN        NaN   
4                 NaN        NaN          NaN        NaN   
5                 NaN        NaN          NaN        NaN   
6                 NaN        NaN          NaN        NaN   
7                 NaN    Caballo          NaN        NaN   
8              Nombre        FEI       Nombre       Raza   
9  Carlos BORHO GIRON    107JQ37  DAMASKUS 57       HANN   

                   Unnamed: 6 Unnamed: 7  
0                         NaN        NaN  
1                         NaN        NaN  
2                         NaN        NaN  
3                         NaN        NaN  
4                         NaN        NaN  
5                         NaN        NaN  
6                         NaN        NaN  
7                  Puntuación     Premio  
8                         NaN        NaN  
9  4/74,63\n12/72,23\n0/36,23          0

In [229]:
def limpieza_puntuacion(elemento):
    for elemento in cosita["Puntuacion"]:
        # print(elemento.split("\n"))
        if pd.isnull(elemento):
            return None, None, None, None, None, None, None, None, None
            
        else:
            celda = elemento.split("\n")
            if len(celda) == 3:
                print(celda)
            elif len(celda) == 2:
            
            elif len(celda) == 1 and "/" not in celda[0]:
                return celda[0], None, None, None, None, None, None, None, None,

['4/74,63', '12/72,23', '0/36,23']
['4/71,87', '8/71,65', '4/36,96']
['0/71,75', '4/68,81', '4/36,28']
['4/67,66', '4/70,91', '0/36,38']
['0/70,42', '4/68,63', '0/35,01']
['4/70,32', '4/71,21', '4/38,08']
['4/72,83', '8/74,32', '6/46,57']
['4/68,61', '8/70,05', '4/34,64']


## Pruebas norma (276.2): Una Manga con Manga Ganadora

In [ ]:
def procesar_puntuacion(puntuacion):
    """
    Procesa la cadena de puntuación para extraer diferentes componentes.

    Args:
        puntuacion (str): La cadena de puntuación a procesar.

    Returns:
        dict: Un diccionario que contiene los componentes extraídos de la puntuación,
              o un diccionario de None si la entrada no es una cadena o está vacía.
    """
    if not isinstance(puntuacion, str) or not puntuacion:
        return {
            "Ptos_Obs_R1": None,
            "Ptos_tiempo_R1": None,
            "Tiempo_R1": None,
            "Ptos_Obs_R2": None,
            "Ptos_tiempo_R2": None,
            "Tiempo_R2": None,
            "Ptos_Obs_R3": None,
            "Ptos_tiempo_R3": None,
            "Tiempo_R3": None,
        }

    partes = puntuacion.split('\n')
    resultados = {
        "Ptos_Obs_R1": None,
        "Ptos_tiempo_R1": None,
        "Tiempo_R1": None,
        "Ptos_Obs_R2": None,
        "Ptos_tiempo_R2": None,
        "Tiempo_R2": None,
        "Ptos_Obs_R3": None,
        "Ptos_tiempo_R3": None,
        "Tiempo_R3": None,
    }

    if len(partes) >= 1:
        if "/" in partes[0]:
            partes_1 = partes[0].split("/")
            if "(" not in partes_1[0]:
                resultados["Ptos_Obs_R1"] = partes_1[0]
                resultados["Tiempo_R1"] = partes_1[1]
            elif "(" in partes_1[0]:
                obs_tiempo = partes_1[0].split("(")
                resultados["Ptos_Obs_R1"] = obs_tiempo[1][0] if len(obs_tiempo) > 1 and len(obs_tiempo[1]) > 0 else None
                resultados["Ptos_tiempo_R1"] = obs_tiempo[1][-6] if len(obs_tiempo) > 1 and len(obs_tiempo[1]) > 6 else None
                resultados["Tiempo_R1"] = partes_1[1]
        else:
            resultados["Ptos_Obs_R1"] = partes[0]

    return resultados

# Crear nuevas columnas inicializadas con None
BA_sc["Ptos_Obs_R1"] = None
BA_sc["Ptos_tiempo_R1"] = None
BA_sc["Tiempo_R1"] = None
BA_sc["Ptos_Obs_R2"] = None
BA_sc["Ptos_tiempo_R2"] = None
BA_sc["Tiempo_R2"] = None
BA_sc["Ptos_Obs_R3"] = None
BA_sc["Ptos_tiempo_R3"] = None
BA_sc["Tiempo_R3"] = None

# Iterar sobre las filas del DataFrame y asignar los valores extraídos a las nuevas columnas
for index, row in BA_sc.iterrows():
    resultados = procesar_puntuacion(row["Puntuacion"])
    BA_sc.loc[index, "Ptos_Obs_R1"] = resultados["Ptos_Obs_R1"]
    BA_sc.loc[index, "Ptos_tiempo_R1"] = resultados["Ptos_tiempo_R1"]
    BA_sc.loc[index, "Tiempo_R1"] = resultados["Tiempo_R1"]
    BA_sc.loc[index, "Ptos_Obs_R2"] = resultados["Ptos_Obs_R2"]
    BA_sc.loc[index, "Ptos_tiempo_R2"] = resultados["Ptos_tiempo_R2"]
    BA_sc.loc[index, "Tiempo_R2"] = resultados["Tiempo_R2"]
    BA_sc.loc[index, "Ptos_Obs_R3"] = resultados["Ptos_Obs_R3"]
    BA_sc.loc[index, "Ptos_tiempo_R3"] = resultados["Ptos_tiempo_R3"]
    BA_sc.loc[index, "Tiempo_R3"] = resultados["Tiempo_R3"]

print(BA_sc)

In [17]:
manga_ganadora = pd.read_excel("../data/data_salto/resultados_2025/RFHE_SALTO_Resultados de la Prueba - 2025-05-13T191013.946.xls")
manga_ganadora = modificaciones_generales(manga_ganadora)
# manga_ganadora[["Puntuacion_1", "Puntuacion_2"]] = manga_ganadora["Puntuacion"].str.split("\n", expand = True)
# manga_ganadora["Puntuacion_3"] = None
# manga_ganadora = manga_ganadora.drop("Puntuacion", axis = 1)
manga_ganadora["Puntuacion"] = manga_ganadora["Puntuacion"].str.split("\n")
print(len(manga_ganadora.iloc[0,-1]))
# for elemento in manga_ganadora["Puntuacion"]:
#     print(len(elemento))
manga_ganadora.head()

TypeError: object of type 'NoneType' has no len()

## Pruebas norma (238.1.1): Baremo A, sin cronómetro
Los igualados a puntos comparten premio y según las condiciones del programa se pueden estipular uno o dos desempates sin cronómetro para los igualados a puntos para el primer puesto.

In [24]:
pd.set_option('display.max_columns', None)

In [234]:
BA_sc = pd.read_excel("../data/data_salto/resultados_2025/RFHE_SALTO_Resultados de la Prueba - 2025-05-13T151603.419.xls")
BA_sc = modificaciones_generales(BA_sc)
# BA_sc["Ptos_Obs_R1"] = BA_sc["Puntuacion"].str.split("/")
# BA_sc["Puntuacion_1"] = BA_sc["Puntuacion"]
# BA_sc["Puntuacion_2"] = None
# BA_sc["Puntuacion_3"] = None
# BA_sc = BA_sc.drop("Puntuacion", axis = 1)
#  coso = BA_sc.iloc[3,9]
# print(coso.split("/")[0].split("(")[1][0])
# print(coso.split("/")[0].split("(")[1][-6])
# print(coso.split("/")[1])

# BA_sc["Puntuacion"] = BA_sc["Puntuacion"].str.split("\n")

# # for elemento in BA_sc["Puntuacion"]:
# #     print(len(elemento))

# celda = BA_sc.iloc[0, -6]
# if len(celda) == 1:

#     if "/" in celda[0]:
#         celda_1 = celda[0].split("/")

#         if "(" not in celda_1:
#             BA_sc["Ptos_Obs_R1"] = BA_sc["Puntuacion"].apply(lambda x: x[0].split("/")[0])
#             BA_sc["Ptos_tiempo_R1"] = None
#             BA_sc["Tiempo_R1"] = BA_sc["Puntuacion"].apply(lambda x: x[0].split("/")[1])

#             BA_sc["Ptos_Obs_R2"] = None
#             BA_sc["Ptos_tiempo_R2"] = None
#             BA_sc["Tiempo_R2"] = None

#             BA_sc["Ptos_Obs_R3"] = None
#             BA_sc["Ptos_tiempo_R3"] = None
#             BA_sc["Tiempo_R3"] = None

#         elif "(" in celda_1:
#             BA_sc["Ptos_Obs_R1"] = BA_sc["Puntuacion"].apply(lambda x: x[0].split("/")[0].split("(")[1][0])
#             BA_sc["Ptos_tiempo_R1"] = BA_sc["Puntuacion"].apply(lambda x: x[0].split("/")[0].split("(")[1][-6])
#             BA_sc["Tiempo_R1"] = BA_sc["Puntuacion"].apply(lambda x: x[0].split("/")[1])

#             BA_sc["Ptos_Obs_R2"] = None
#             BA_sc["Ptos_tiempo_R2"] = None
#             BA_sc["Tiempo_R2"] = None

#             BA_sc["Ptos_Obs_R3"] = None
#             BA_sc["Ptos_tiempo_R3"] = None
#             BA_sc["Tiempo_R3"] = None
#         else:
#             pass

#     elif "/" not in celda[0]:

#         BA_sc["Ptos_Obs_R1"] = BA_sc["Puntuacion"].apply(lambda x: x[0])
#         BA_sc["Ptos_tiempo_R1"] = None
#         BA_sc["Tiempo_R1"] = None

#         BA_sc["Ptos_Obs_R2"] = None
#         BA_sc["Ptos_tiempo_R2"] = None
#         BA_sc["Tiempo_R2"] = None

#         BA_sc["Ptos_Obs_R3"] = None
#         BA_sc["Ptos_tiempo_R3"] = None
#         BA_sc["Tiempo_R3"] = None
# BA_sc["Puntuacion"] = BA_sc["Puntuacion"].str.split("\n")
BA_sc
# BA_sc.iloc[3,9][0].split("/")[0].split("(")[1][-6]

Estado  Posicion Lic_jinete                            Jinete Pais_jinete  \
0     FIN         1   10040364              Luis FRANCO MARTÍNEZ         ESP   
1     FIN         2   10116527                Giorgia DE LORENZI         ITA   
2     FIN         3   10005772                      Kevin GIELEN         BEL   
3     FIN         4   10289081  Christina Elisabeth Norgaard RUD         NOR   
4     FIN         5   10104173                    Valeria BIAZZI         ITA   
5     FIN         6   10139698                      Ronnie JONES         GBR   
6     FIN         7   10025853                     Gilles BOTTON         BEL   
7     FIN         8   10138266                   Scott DOLLEMORE         GBR   
8     FIN         9   10114636               Fidel DÁVILA GARCÍA         ESP   
9     FIN        10   10048768                    Emilie PAILLOT         SUI   
10    FIN        11   10003772              Carlos BOSCH CEBRIÁN         ESP   
11    FIN        12   10001090                     Davide FENILI         ITA   
12    FIN        13   10011791                 Dominique JOASSIN         BEL   
13    FIN        14   10031696                  Katharina WERNDL         GER   
14    FIN        15   10041101                    Laura ROBINSON         GBR   
15    FIN        16   10257328              Manuel ROMÁN ÁLVAREZ         ESP   
16    FIN        17   10174768               Giacomo LANZI MOTTO         ESP   
17    FIN        18   10032766  Jesús BAMONDE BERMÚDEZ DE CASTRO         ESP   
18    FIN        19   10366744                   Amadeo SALSETTA         BEL   

   Lic_caballo                       Caballo Raza_caballo Pais_caballo  \
0      109DA71                FENIX ANANTARA          CDE          ESP   
1      109CB95                      BADDALOU         None          ITA   
2      109BF73    CHARLY VAN HET CAUTERHOF Z         None          BEL   
3      109AW75                    UNICO DOMM         None          NOR   
4      109CC29                BRIGITTE BELLE         None          ITA   
5      109BS53             SLH ICAN DOMINATE         None          GBR   
6      109AN17  COMMILF'OH DE LA MARCHETTE Z         None          BEL   
7      109CJ54                   BEA BLUE PS         None          GBR   
8      109CI39               KAN DE LLAMOSAS           SF          ESP   
9      109BY27                      PRADA ES         None          SUI   
10     109CP94             ARETHA DE MONTERO          CDE          ESP   
11     109CP53                   HALLEY GREY         None          ITA   
12     109BZ64                DARK-LOVE LJ Z         None          BEL   
13     109AD51         POSEIDON VAN DE VIHTA         None          GER   
14     109AU63            KAOLIN DE BEAUVOIR         None          GBR   
15     109BS27              KALOURINA D ODEA          CDE          ESP   
16     109BN55                    PILLOT BMG         KWPN          ESP   
17     109CQ38        PARSIFAL DE ENCINASOLA          CDE          ESP   
18     109CB36          OLOGIE DU PONT-RIHEN         None          BEL   

                    Puntuacion  Premio Dinero_premio  \
0                      0/74,05    True            75   
1                      0/76,34    True            60   
2                      0/77,69    True            45   
3       4(4 Obs - 0 Tpo)/70,88    True            30   
4       4(4 Obs - 0 Tpo)/73,38    True            21   
5       4(4 Obs - 0 Tpo)/75,42    True          16.5   
6       4(4 Obs - 0 Tpo)/75,48    True            12   
7       8(8 Obs - 0 Tpo)/73,56    True             9   
8       8(8 Obs - 0 Tpo)/74,43    True             9   
9       8(8 Obs - 0 Tpo)/79,54    True           7.5   
10      8(8 Obs - 0 Tpo)/83,57    True           7.5   
11      9(8 Obs - 1 Tpo)/84,54    True           7.5   
12    12(12 Obs - 0 Tpo)/68,69   False             0   
13    12(12 Obs - 0 Tpo)/75,16   False             0   
14    12(12 Obs - 0 Tpo)/79,10   False             0   
15    16(12 Obs - 4 Tpo)/

In [153]:
for indice, row in BA_sc.iterrows():
    celda = row["Puntuacion"]
        
    if len(celda) == 1:
        puntuacion = celda[0] # accedemos al primer elemento de la lista, que en este caso tiene solo un elemento, el 0
        # print(puntuacion)

        if "/" in puntuacion:
            partes = puntuacion.split("/")
            # print(partes[0])

            if "(" not in partes[0]:
                BA_sc.at[indice, "Ptos_Obs_R1"] = partes[0]
                BA_sc.at[indice, "Tiempo_R1"] = partes[1]
            elif "(" in partes[0]:
                ptos_obs = partes[0].split("(")[1].split("-")[0][:-5].strip()
                ptos_tiempo = partes[0].split("(")[1].split("-")[1][:-5].strip()
                BA_sc.at[indice, "Ptos_Obs_R1"] = ptos_obs
                BA_sc.at[indice, "Ptos_tiempo_R1"] = ptos_tiempo
                BA_sc.at[indice, "Tiempo_R1"] = partes[1]
        else:
            BA_sc.at[indice, "Ptos_Obs_R1"] = puntuacion

    elif len(celda) == 2:
        puntuacion_r1 = celda[0]
        puntuacion_r2 = celda[1]

        if "/" in puntuacion_r1:
            partes_r1 = puntuacion_r1.split("/")
            if "(" not in partes_r1[0]:
                BA_sc.at[indice, "Ptos_Obs_R1"] = partes_r1[0]
                BA_sc.at[indice, "Tiempo_R1"] = partes_r1[1]

            elif "(" in partes_r1[0]:
                ptos_obs_r1 = partes_r1[0].split("(")[1].split("-")[0][:-5].strip()
                ptos_tiempo_r1 = partes_r1[0].split("(")[1].split("-")[1][:-5].strip()
                BA_sc.at[indice, "Ptos_Obs_R1"] = ptos_obs_r1
                BA_sc.at[indice, "Ptos_tiempo_R1"] = ptos_tiempo_r1
                BA_sc.at[indice, "Tiempo_R1"] = partes_r1[1]
        else:
            BA_sc.at[indice, "Ptos_Obs_R1"] = puntuacion_r1
        
        if "/" in puntuacion_r2:
            partes_r2 = puntuacion_r2.split("/")
            if "(" not in partes_r2[0]:
                BA_sc.at[indice, "Ptos_Obs_R2"] = partes_r2[0]
                BA_sc.at[indice, "Tiempo_R2"] = partes_r2[1]

            elif "(" in partes_r2[0]:
                ptos_obs_r2 = partes_r2[0].split("(")[1].split("-")[0][:-5].strip()
                ptos_tiempo_r2 = partes_r2[0].split("(")[1].split("-")[1][:-5].strip()
                BA_sc.at[indice, "Ptos_Obs_R2"] = ptos_obs_r2
                BA_sc.at[indice, "Ptos_tiempo_R2"] = ptos_tiempo_r2
                BA_sc.at[indice, "Tiempo_R2"] = partes_r2[1]
        else:
            BA_sc.at[indice, "Ptos_Obs_R2"] = puntuacion_r2


BA_sc.head(15)

Estado  Posicion Lic_jinete                            Jinete Pais_jinete  \
0     FIN         1   10040364              Luis FRANCO MARTÍNEZ         ESP   
1     FIN         2   10116527                Giorgia DE LORENZI         ITA   
2     FIN         3   10005772                      Kevin GIELEN         BEL   
3     FIN         4   10289081  Christina Elisabeth Norgaard RUD         NOR   
4     FIN         5   10104173                    Valeria BIAZZI         ITA   
5     FIN         6   10139698                      Ronnie JONES         GBR   
6     FIN         7   10025853                     Gilles BOTTON         BEL   
7     FIN         8   10138266                   Scott DOLLEMORE         GBR   
8     FIN         9   10114636               Fidel DÁVILA GARCÍA         ESP   
9     FIN        10   10048768                    Emilie PAILLOT         SUI   
10    FIN        11   10003772              Carlos BOSCH CEBRIÁN         ESP   
11    FIN        12   10001090                     Davide FENILI         ITA   
12    FIN        13   10011791                 Dominique JOASSIN         BEL   
13    FIN        14   10031696                  Katharina WERNDL         GER   
14    FIN        15   10041101                    Laura ROBINSON         GBR   

   Lic_caballo                       Caballo Raza_caballo Pais_caballo  \
0      109DA71                FENIX ANANTARA          CDE          ESP   
1      109CB95                      BADDALOU         None          ITA   
2      109BF73    CHARLY VAN HET CAUTERHOF Z         None          BEL   
3      109AW75                    UNICO DOMM         None          NOR   
4      109CC29                BRIGITTE BELLE         None          ITA   
5      109BS53             SLH ICAN DOMINATE         None          GBR   
6      109AN17  COMMILF'OH DE LA MARCHETTE Z         None          BEL   
7      109CJ54                   BEA BLUE PS         None          GBR   
8      109CI39               KAN DE LLAMOSAS           SF          ESP   
9      109BY27                      PRADA ES         None          SUI   
10     109CP94             ARETHA DE MONTERO          CDE          ESP   
11     109CP53                   HALLEY GREY         None          ITA   
12     109BZ64                DARK-LOVE LJ Z         None          BEL   
13     109AD51         POSEIDON VAN DE VIHTA         None          GER   
14     109AU63            KAOLIN DE BEAUVOIR         None          GBR   

                    Puntuacion  Premio Dinero_premio  \
0                    [0/74,05]    True            75   
1                    [0/76,34]    True            60   
2                    [0/77,69]    True            45   
3     [4(4 Obs - 0 Tpo)/70,88]    True            30   
4     [4(4 Obs - 0 Tpo)/73,38]    True            21   
5     [4(4 Obs - 0 Tpo)/75,42]    True          16.5   
6     [4(4 Obs - 0 Tpo)/75,48]    True            12   
7     [8(8 Obs - 0 Tpo)/73,56]    True             9   
8     [8(8 Obs - 0 Tpo)/74,43]    True             9   
9     [8(8 Obs - 0 Tpo)/79,54]    True           7.5   
10    [8(8 Obs - 0 Tpo)/83,57]    True           7.5   
11    [9(8 Obs - 1 Tpo)/84,54]    True           7.5   
12  [12(12 Obs - 0 Tpo)/68,69]   False             0   
13  [12(12 Obs - 0 Tpo)/75,16]   False             0   
14  [12(12 Obs - 0 Tpo)/79,10]   False             0   

                                        Prueba Fecha_prueba  \
0   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
1   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
2   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
3   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
4   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
5   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
6   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
7   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
8   5 años - 1,15 - A sin cronómetro (238.1.1)   2025-02-19   
9   5 años

## Pruebas norma (239): Baremo C
Las faltas son penalizadas en segundos que se agregan al tiempo empleado por el aprticipante en completar el recorrido o eliminación. El resultado se determinará como la suma del tiempo del recorrido incluidas las correcciones de tiempo, si las hubiera, más cuatro segundos por obstáculo derribado (tres segundos en desempates según baremo C o en la segunda fase de la prueba de dos fases o en eliminatorias sucesivas) da el resultado en segundos, para el participante.
- RFHE_SALTO_Resultados de la Prueba - 2025-05-13T192941.693.xls
- RFHE_SALTO_Resultados de la Prueba - 2025-05-13T193001.145.xls

In [95]:
pd.set_option('display.max_columns', None)


In [35]:
BC = pd.read_excel("../data/data_salto/resultados_2025/RFHE_SALTO_Resultados de la Prueba - 2025-05-13T193001.145.xls")
BC = modificaciones_generales(BC)
# BC["Puntuacion_1"] = BC["Puntuacion"].str.split("\n", expand = True)
# BC["Ptos_Obs_R1"] = BC["Puntuacion"].apply(lambda x: x.split("/")[0])
# BC["Puntuacion"] = BC["Puntuacion"].str.split("\n")

# if len(BC.iloc[0, -6]) == 1 and "/" in BC.iloc[0, -6][0]:
#     BC["Ptos_Obs_R1"] = BC["Puntuacion"].apply(lambda x: x[0].split("/")[0])
#     BC["Ptos_tiempo_R1"] = None
#     BC["Tiempo_R1"] = None

#     BC["Ptos_Obs_R2"] = None
#     BC["Ptos_tiempo_R2"] = None
#     BC["Tiempo_R2"] = None

#     BC["Ptos_Obs_R3"] = None
#     BC["Ptos_tiempo_R3"] = None
#     BC["Tiempo_R3"] = None
    
# elif len(BC.iloc[0, -6]) == 1 and "/" not in BC.iloc[0, -6][0]:
    
#     BC["Ptos_Obs_R1"] = BC["Puntuacion"].apply(lambda x: x[0])
#     BC["Ptos_tiempo_R1"] = None
#     BC["Tiempo_R1"] = None

#     BC["Ptos_Obs_R2"] = None
#     BC["Ptos_tiempo_R2"] = None
#     BC["Tiempo_R2"] = None

#     BC["Ptos_Obs_R3"] = None
#     BC["Ptos_tiempo_R3"] = None
#     BC["Tiempo_R3"] = None
# # len(BC.iloc[0,-1])
# BC["Puntuacion_2"] = None
# BC["Puntuacion_3"] = None
# BC = BC.drop("Puntuacion", axis = 1)
BC["Puntuacion"] = BC["Puntuacion"].str.split("\n")
BC

Estado  Posicion Lic_jinete                        Jinete Pais_jinete  \
0    FIN         1      45827          Luna JIMÉNEZ BECHOLD         ESP   
1    FIN         2      70254      Alejandra MELIAN SANTANA         ESP   
2    FIN         3      45827          Luna JIMÉNEZ BECHOLD         ESP   
3    FIN         4     155208           Loic MOLINA MÁRQUEZ         ESP   
4    FIN         5     273709  Saul Alejandro SÁNCHEZ ROJAS         ESP   
5    FIN         6      38323                Nicole STAIGER         ESP   
6    FIN         7      33381         Isabel PALLERO DÁVILA         ESP   

  Lic_caballo                                  Caballo Raza_caballo  \
0      165666  SUPREME DU MILON MERCADO DEL AGRICULTOR           SF   
1      167713                             IL FOTOGRAFO          PSI   
2      162922                  OHIO LV VAN T BIEZENHOF          BWP   
3      155995                                  GWYNETH         KWPN   
4      168850                                   GIMORA         KWPN   
5      146915                                BENALUP C          CDE   
6       90270                         MARISCAL DEL PAS          CDE   

  Pais_caballo Puntuacion  Premio Dinero_premio             Prueba  \
0          ESP    [79,10]   False             0  1,30 - Caza (239)   
1          ESP    [80,58]   False             0  1,30 - Caza (239)   
2          ESP    [92,42]   False             0  1,30 - Caza (239)   
3          ESP    [96,75]   False             0  1,30 - Caza (239)   
4          ESP   [101,53]   False             0  1,30 - Caza (239)   
5          ESP   [105,58]   False             0  1,30 - Caza (239)   
6          ESP   [115,45]   False             0  1,30 - Caza (239)   

  Fecha_prueba                                           Concurso Ptos_Obs_R1  \
0   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...        None   
1   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...        None   
2   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...        None   
3   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...        None   
4   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...        None   
5   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...        None   
6   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...        None   

  Ptos_tiempo_R1 Tiempo_R1 Ptos_Obs_R2 Ptos_tiempo_R2 Tiempo_R2 Ptos_Obs_R3  \
0           None      None        None           None      None        None   
1           None      None        None           None      None        None   
2           None      None        None           None      None        None   
3           None      None        None           None      None        None   
4           None      None        None           None      None        None   
5           None      None        None           None      None        None   
6           None      None        None           None      None        None   

  Ptos_tiempo_R3 Tiempo_R3  
0           None      None  
1           None      None  
2           None      None  
3           None      None  
4           None      None  
5           None      None  
6           None      None

In [99]:
for indice, row in BC.iterrows():
    celda = row["Puntuacion"]
        
    if len(celda) == 1:
        puntuacion = celda[0] # accedemos al primer elemento de la lista, que en este caso tiene solo un elemento, el 0
        # print(puntuacion)

        if "/" in puntuacion:
            partes = puntuacion.split("/")
            # print(partes[0])

            if "(" not in partes[0]:
                BC.at[indice, "Ptos_Obs_R1"] = partes[0]
                BC.at[indice, "Tiempo_R1"] = partes[1]
            else:
                ptos_obs = partes[0].split("(")[1][0]
                ptos_tiempo = partes[0].split("(")[1][-6]
                BC.at[indice, "Ptos_Obs_R1"] = ptos_obs
                BC.at[indice, "Ptos_tiempo_R1"] = ptos_tiempo
                BC.at[indice, "Tiempo_R1"] = partes[1]
        else:
            BC.at[indice, "Ptos_Obs_R1"] = puntuacion
            
BC.head(8)

Estado  Posicion Lic_jinete                        Jinete Pais_jinete  \
0    FIN         1      45827          Luna JIMÉNEZ BECHOLD         ESP   
1    FIN         2      70254      Alejandra MELIAN SANTANA         ESP   
2    FIN         3      45827          Luna JIMÉNEZ BECHOLD         ESP   
3    FIN         4     155208           Loic MOLINA MÁRQUEZ         ESP   
4    FIN         5     273709  Saul Alejandro SÁNCHEZ ROJAS         ESP   
5    FIN         6      38323                Nicole STAIGER         ESP   
6    FIN         7      33381         Isabel PALLERO DÁVILA         ESP   

  Lic_caballo                                  Caballo Raza_caballo  \
0      165666  SUPREME DU MILON MERCADO DEL AGRICULTOR           SF   
1      167713                             IL FOTOGRAFO          PSI   
2      162922                  OHIO LV VAN T BIEZENHOF          BWP   
3      155995                                  GWYNETH         KWPN   
4      168850                                   GIMORA         KWPN   
5      146915                                BENALUP C          CDE   
6       90270                         MARISCAL DEL PAS          CDE   

  Pais_caballo Puntuacion  Premio Dinero_premio             Prueba  \
0          ESP    [79,10]   False             0  1,30 - Caza (239)   
1          ESP    [80,58]   False             0  1,30 - Caza (239)   
2          ESP    [92,42]   False             0  1,30 - Caza (239)   
3          ESP    [96,75]   False             0  1,30 - Caza (239)   
4          ESP   [101,53]   False             0  1,30 - Caza (239)   
5          ESP   [105,58]   False             0  1,30 - Caza (239)   
6          ESP   [115,45]   False             0  1,30 - Caza (239)   

  Fecha_prueba                                           Concurso Ptos_Obs_R1  \
0   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...       79,10   
1   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...       80,58   
2   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...       92,42   
3   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...       96,75   
4   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...      101,53   
5   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...      105,58   
6   2025-04-19  CSN* C.H. DEL SUR - SEMANA SANTA (17/04/2025 -...      115,45   

  Ptos_tiempo_R1 Tiempo_R1 Ptos_Obs_R2 Ptos_tiempo_R2 Tiempo_R2 Ptos_Obs_R3  \
0           None      None        None           None      None        None   
1           None      None        None           None      None        None   
2           None      None        None           None      None        None   
3           None      None        None           None      None        None   
4           None      None        None           None      None        None   
5           None      None        None           None      None        None   
6           None      None        None           None      None        None   

  Ptos_tiempo_R3 Tiempo_R3  
0           None      None  
1           None      None  
2           None      None  
3           None      None  
4           None      None  
5           None      None  
6           None      None

## Pruebas norma (273.3.3.1): Prueba en dos Mangas

Ambas mangas en Baremo A, ambas con cronómetro.

- RFHE_SALTO_Resultados de la Prueba - 2025-05-13T155509.489.xls
- RFHE_SALTO_Resultados de la Prueba - 2025-05-13T163100.768.xls
- RFHE_SALTO_Resultados de la Prueba - 2025-05-13T185621.773.xls
- RFHE_SALTO_Resultados de la Prueba - 2025-05-13T192327.719.xls

In [236]:
Dos_m = pd.read_excel("../data/data_salto/resultados_2025/RFHE_SALTO_Resultados de la Prueba - 2025-05-13T155509.489.xls")
Dos_m = modificaciones_generales(Dos_m)
Dos_m
# Dos_m["Puntuacion"] = Dos_m["Puntuacion"].str.split("\n")
# # Dos_m
# Dos_m.iloc[-10,9][0].split("/")[0].split("(")[1].split("-")[1][:-5].strip()

Estado Posicion Lic_jinete                           Jinete Pais_jinete  \
0     FIN        1       5042  Pedro Fernando MATEOS RODRÍGUEZ         ESP   
1     FIN        2     274244                Alex CODINA LERIA         ESP   
2     FIN        3     132635           Pilar MATEOS DE URBINA         ESP   
3     FIN        4     369989       Emmy Pauline KROGER KROGER         ESP   
4     FIN        5     323398               Alex TEJADA BEROIZ         ESP   
5     FIN        6      15975  Alvaro SANCHEZ-ARCILLA ARTIÑANO         ESP   
6     FIN        7       5018     Javier José MATEOS RODRÍGUEZ         ESP   
7     FIN        8      14425      Carmen GARCIA-TORRES SOLANO         ESP   
8     FIN        9     323398               Alex TEJADA BEROIZ         ESP   
9     FIN       10      14425      Carmen GARCIA-TORRES SOLANO         ESP   
10    FIN       11      68788         Ana GARCIA-TORRES SOLANO         ESP   
11    FIN       12     198045        Rodrigo MARTÍN GARAIZABAL         ESP   
12    FIN       13     315225           Gonzalo BEDOYA AGUILAR         ESP   
13    FIN       14      16235              Luis MÁRQUEZ MÉNDEZ         ESP   
14    FIN       15     135856     Joaquín TINAO PEREZ-MIRAVETE         ESP   
15    FIN       16     135856     Joaquín TINAO PEREZ-MIRAVETE         ESP   
16    FIN       17     333368           Emma RODRIGUEZ VAZQUEZ         ESP   
17    FIN       18     196126       Francisco ROMÁN CANTILLANA         ESP   
18    FIN       19      97878              Iñaki LORENZO SIMÓN         ESP   
19    FIN       20     272908           Angel CERDIDO CALDERON         ESP   
20    FIN       21     201186           Salvador ASTOLFI NÚÑEZ         ESP   
21    FIN       22      16235              Luis MÁRQUEZ MÉNDEZ         ESP   
22    FIN       23     274244                Alex CODINA LERIA         ESP   
23    FIN       24      36359    Fernando DE LOS SANTOS GARCÍA         ESP   
24    FIN       25      46311               Miguel LAIZ ZANDIO         ESP   
25    FIN       26      36359    Fernando DE LOS SANTOS GARCÍA         ESP   
26    FIN       27     165303     Lucas DE HEREDIA MUÑOZ-ROJAS         ESP   
27    FIN       28      32156                Miguel ARES GÓMEZ         ESP   
28    FIN       29      17253          Antonio BENITO BLÁZQUEZ         ESP   
29    RET      RET     273669              Alvaro CASTRO ROMÁN         ESP   
30    RET      RET       5042  Pedro Fernando MATEOS RODRÍGUEZ         ESP   
31    ELI      ELI      60198          Marta BARTOLOMÉ REDONDO         ESP   
32    ELI      ELI     132635           Pilar MATEOS DE URBINA         ESP   
33    ELI      ELI     319800               Lucas MOROTE SORIA         ESP   
34    NOP      NOP     133027            Celine ARAGÓN LORENZO         ESP   

   Lic_caballo                      Caballo Raza_caballo Pais_caballo  \
0       149266                    QUIZAS CC          CDE          ESP   
1       155917                  COSTELLO DC          CDE          ESP   
2       147592             AZURE DES ROCHES           SF          ESP   
3       170700                DSP DACENTO H        REITF          ESP   
4       116440            APOLO DE EQUIANPE          CDE          ESP   
5       156003                   CHAMP BLUE        REITF          ESP   
6       143177                      LEVIS C         HANN          ESP   
7       159657              FOR JEWEL EDM Z          BWP          ESP   
8       126325               TABATA DE OLID          CDE          ESP   
9       159656           QUOTE FOR PLEASURE          BWP          ESP   
10      137827         ELTON BLUE D'ARGONNE           SF          ESP   
11      141281              PICASSO DEL ARA          CDE          ESP   
12      148938                   ZEUS MAG Z         ZANG          ESP   
13      138817            ROTTERDAM BUZALEN          CDE          ESP   
14      156938                   OLLIVER SW          SBS          ESP   
15      153827  ZOE BLUE V

In [235]:

Dos_m

Unnamed: 0                                         Unnamed: 1  \
0          NaN                                                NaN   
1    Concurso:  CSI4* - 5th Week XXXI Andalucía Sunshine Tour ...   
2       Fecha:                                2025-03-08 00:00:00   
3      Prueba:                   1,45 (GP) - Dos Mangas (273.3.4)   
4   Categoría:                                            CSI****   
..         ...                                                ...   
65         RET                                           10071629   
66         RET                                           10137362   
67         RET                                           10041949   
68         RET                                           10002578   
69         ELI                                           10056071   

               Unnamed: 2 Unnamed: 3                    Unnamed: 4 Unnamed: 5  \
0                     NaN        NaN                           NaN        NaN   
1                     NaN        NaN                           NaN        NaN   
2                     NaN        NaN                           NaN        NaN   
3                     NaN        NaN                           NaN        NaN   
4                     NaN        NaN                           NaN        NaN   
..                    ...        ...                           ...        ...   
65      Nicola POHL (GER)    105LA15         ARLO DE BLONDEL (GBR)         --   
66    Emma CRAWFORD (GBR)    107DV35        DUKES OF HAZZARD (GBR)         --   
67  Andrej PAVLOVIC (SLO)    108ME31                    KING (NED)         --   
68    Radovan SILLO (SVK)    107KT56  CARLO VAN KLAPSCHEUT Z (BEL)         --   
69  Jacob HELLSTRÖM (SWE)    107AN71                 CHARLES (SWE)         --   

   Unnamed: 6 Unnamed: 7  
0         NaN        NaN  
1         NaN        NaN  
2         NaN        NaN  
3         NaN        NaN  
4         NaN        NaN  
..        ...        ...  
65        RET          0  
66        RET          0  
67        RET          0  
68        RET          0  
69        ELI          0  

[70 rows x 8 columns]

## Pruebas norma (273.3.4): Prueba en dos Mangas

Ambas mangas en Baremo A, ambas con cronómetro.

In [152]:
Dos_m = pd.read_excel("../data/data_salto/resultados_2025/RFHE_SALTO_Resultados de la Prueba - 2025-05-13T165920.069.xls")
# Dos_m = modificaciones_generales(Dos_m)
# Dos_m[["Puntuacion_1", "Puntuacion_2"]] = Dos_m["Puntuacion"].str.split("\n", expand = True)
# Dos_m["Puntuacion_3"] = None
# # Dos_m = Dos_m.drop("Puntuacion", axis = 1)
# Dos_m["Puntuacion"] = Dos_m["Puntuacion"].str.split("\n")
Dos_m.head(20)

Unnamed: 0                                         Unnamed: 1  \
0          NaN                                                NaN   
1    Concurso:  CSI4* - 5th Week XXXI Andalucía Sunshine Tour ...   
2       Fecha:                                2025-03-08 00:00:00   
3      Prueba:                   1,45 (GP) - Dos Mangas (273.3.4)   
4   Categoría:                                            CSI****   
5          NaN                                                NaN   
6          NaN                                                NaN   
7       Puesto                                     Jinete/Amazona   
8          NaN                                                FEI   
9            1                                           10016473   
10           2                                           10150533   
11           3                                           10045115   
12           4                                           10058729   
13           5                                           10097112   
14           6                                           10002057   
15           7                                           10017385   
16           8                                           10098599   
17           9                                           10185632   
18          10                                           10005788   
19          11                                           10000452   

                         Unnamed: 2 Unnamed: 3                    Unnamed: 4  \
0                               NaN        NaN                           NaN   
1                               NaN        NaN                           NaN   
2                               NaN        NaN                           NaN   
3                               NaN        NaN                           NaN   
4                               NaN        NaN                           NaN   
5                               NaN        NaN                           NaN   
6                               NaN        NaN                           NaN   
7                               NaN    Caballo                           NaN   
8                            Nombre        FEI                        Nombre   
9          Penelope LEPREVOST (FRA)    106VL82         DJAGGER SEMILLY (FRA)   
10             Finn BOEREKAMP (NED)    106RM98                    JAY JAY ES   
11          Anthony BOURQUARD (SUI)    108CC64        FLANAGAN SEMILLY (SUI)   
12  Rodrigo Giesteira ALMEIDA (POR)    107DF31          COMME LE COEUR (GER)   
13             Sanne THIJSSEN (NED)    107FW98             LOLANDRIA Z (NED)   
14          Luca Maria MONETA (ITA)    108CQ45            EL CAPONE RK (FRA)   
15           William WHITAKER (GBR)    107HD18     MILLFIELD QUICKSTEP (GBR)   
16           William FLETCHER (GBR)    108PH34  CALIXTE HEARTBREAKER Z (GBR)   
17          Anastasia NIELSEN (MON)    105VW34       TAILORMADE CONTOU (DEN)   
18          Camila MAZZA SMIT (BRA)    107HA61  BARBAROSSA DREE BOEKEN (BEL)   
19              Martin DOPAZO (ARG)    107IK37                    A.N.BARNEY   

   Unnamed: 5  Unnamed: 6 Unnamed: 7  
0         NaN         NaN        NaN  
1         NaN         NaN        NaN  
2         NaN         NaN        NaN  
3         NaN         NaN        NaN  
4         NaN         NaN        NaN  
5         NaN         NaN        NaN  
6         NaN         NaN        NaN  
7         NaN  Puntuación     Premio  
8        Raza         NaN        NaN  
9          --     0/59,15       7050  
10       KWPN     0/59,68       5640  
11         --     0/60,28       4230  
12         --     0/60,49       2820  
13         --     0/61,21       1974  
14         --     0/62,35       1551  
15         --     0/64,63       1128  
16         --     0/64,75        846  
17         --     0/64,81        846  
18         --     0/66,38        705  
19        DWB     0/66,39        705

In [149]:
for indice, row in Dos_m.iterrows():
    celda = row["Puntuacion"]
        
    if len(celda) == 1:
        puntuacion = celda[0] # accedemos al primer elemento de la lista, que en este caso tiene solo un elemento, el 0
        # print(puntuacion)

        if "/" in puntuacion:
            partes = puntuacion.split("/")
            # print(partes[0])

            if "(" not in partes[0]:
                Dos_m.at[indice, "Ptos_Obs_R1"] = partes[0]
                Dos_m.at[indice, "Tiempo_R1"] = partes[1]
            elif "(" in partes[0]:
                ptos_obs = partes[0].split("(")[1].split("-")[0][:-5].strip()
                ptos_tiempo = partes[0].split("(")[1].split("-")[1][:-5].strip()
                Dos_m.at[indice, "Ptos_Obs_R1"] = ptos_obs
                Dos_m.at[indice, "Ptos_tiempo_R1"] = ptos_tiempo
                Dos_m.at[indice, "Tiempo_R1"] = partes[1]
        else:
            Dos_m.at[indice, "Ptos_Obs_R1"] = puntuacion

    elif len(celda) == 2:
        puntuacion_r1 = celda[0]
        puntuacion_r2 = celda[1]

        if "/" in puntuacion_r1:
            partes_r1 = puntuacion_r1.split("/")
            if "(" not in partes_r1[0]:
                Dos_m.at[indice, "Ptos_Obs_R1"] = partes_r1[0]
                Dos_m.at[indice, "Tiempo_R1"] = partes_r1[1]

            elif "(" in partes_r1[0]:
                ptos_obs_r1 = partes_r1[0].split("(")[1].split("-")[0][:-5].strip()
                ptos_tiempo_r1 = partes_r1[0].split("(")[1].split("-")[1][:-5].strip()
                Dos_m.at[indice, "Ptos_Obs_R1"] = ptos_obs_r1
                Dos_m.at[indice, "Ptos_tiempo_R1"] = ptos_tiempo_r1
                Dos_m.at[indice, "Tiempo_R1"] = partes_r1[1]
        else:
            BA_sc.at[indice, "Ptos_Obs_R1"] = puntuacion_r1
        
        if "/" in puntuacion_r2:
            partes_r2 = puntuacion_r2.split("/")
            if "(" not in partes_r2[0]:
                Dos_m.at[indice, "Ptos_Obs_R2"] = partes_r2[0]
                Dos_m.at[indice, "Tiempo_R2"] = partes_r2[1]

            elif "(" in partes_r2[0]:
                ptos_obs_r2 = partes_r2[0].split("(")[1].split("-")[0][:-5].strip()
                ptos_tiempo_r2 = partes_r2[0].split("(")[1].split("-")[1][:-5].strip()
                Dos_m.at[indice, "Ptos_Obs_R2"] = ptos_obs_r2
                Dos_m.at[indice, "Ptos_tiempo_R2"] = ptos_tiempo_r2
                Dos_m.at[indice, "Tiempo_R2"] = partes_r2[1]
        else:
            Dos_m.at[indice, "Ptos_Obs_R2"] = puntuacion_r2


Dos_m.head(15)

Estado Posicion Lic_jinete                     Jinete Pais_jinete  \
0     FIN        1   10016473         Penelope LEPREVOST         FRA   
1     FIN        2   10150533             Finn BOEREKAMP         NED   
2     FIN        3   10045115          Anthony BOURQUARD         SUI   
3     FIN        4   10058729  Rodrigo Giesteira ALMEIDA         POR   
4     FIN        5   10097112             Sanne THIJSSEN         NED   
5     FIN        6   10002057          Luca Maria MONETA         ITA   
6     FIN        7   10017385           William WHITAKER         GBR   
7     FIN        8   10098599           William FLETCHER         GBR   
8     FIN        9   10185632          Anastasia NIELSEN         MON   
9     FIN       10   10005788          Camila MAZZA SMIT         BRA   
10    FIN       11   10000452              Martin DOPAZO         ARG   
11    FIN       12   10001206          Emiliano LIBERATI         ITA   
12    FIN       13   10077490                 Emily WARD         GBR   
13    FIN       14   10210656      Pablo SANTIAGO ARTIME         ESP   
14    FIN       15   10001975                László TÓTH         HUN   

   Lic_caballo                 Caballo Raza_caballo Pais_caballo Puntuacion  \
0      106VL82         DJAGGER SEMILLY         None          FRA  [0/59,15]   
1      106RM98              JAY JAY ES         KWPN          ESP  [0/59,68]   
2      108CC64        FLANAGAN SEMILLY         None          SUI  [0/60,28]   
3      107DF31          COMME LE COEUR         None          GER  [0/60,49]   
4      107FW98             LOLANDRIA Z         None          NED  [0/61,21]   
5      108CQ45            EL CAPONE RK         None          FRA  [0/62,35]   
6      107HD18     MILLFIELD QUICKSTEP         None          GBR  [0/64,63]   
7      108PH34  CALIXTE HEARTBREAKER Z         None          GBR  [0/64,75]   
8      105VW34       TAILORMADE CONTOU         None          DEN  [0/64,81]   
9      107HA61  BARBAROSSA DREE BOEKEN         None          BEL  [0/66,38]   
10     107IK37              A.N.BARNEY          DWB          ESP  [0/66,39]   
11     105EA49               JAPPELOUP         None          BEL  [0/66,54]   
12     107KC96            WILD STAR HF         None          GBR  [0/67,33]   
13     106IC24  CHIQUITA DE D'ALGUILLY           SF          ESP  [0/67,76]   
14     108ZM13            TOLL CENTRAL         None          HUN  [0/68,58]   

    Premio Dinero_premio                            Prueba Fecha_prueba  \
0     True          7050  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
1     True          5640  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
2     True          4230  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
3     True          2820  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
4     True          1974  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
5     True          1551  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
6     True          1128  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
7     True           846  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
8     True           846  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
9     True           705  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
10    True           705  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
11    True           705  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
12    True           100  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
13    True           100  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   
14    True           100  1,45 (GP) - Dos Mangas (273.3.4)   2025-03-08   

                                             Concurso Ptos_Obs_R1  \
0   CSI4* - 5th Week XXXI Andalucía Sunshine Tour ...           0   
1   CSI4* - 5th Week XXXI Andalucía Sunshine Tour ...           0   
2   CSI4* - 5th Week XXXI Andalucía Sunshine Tour ...           0   
3   CSI4* - 5th Week XXXI Andalucía Sunshine Tour ...           0   
4   CSI4* - 5th Week XXXI Andalucía Sun

# Valores jsons

In [3]:
import json

# Cargar el JSON
with open('../data/data_salto/concursos/concursos_salto_internacional_2025.json', 'r', encoding='utf-8') as file:
    datos_concursos = json.load(file)

# Obtener la longitud de referencia
longitud_maxima = max(len(v) for v in datos_concursos.values() if isinstance(v, list))

# Listas a revisar
listas_objetivo = ["Inicio", "Final", "Ambito"]

# Detectar y mostrar valores faltantes
for key in listas_objetivo:
    lista = datos_concursos[key]
    indices_presentes = set(range(len(lista)))
    indices_totales = set(range(longitud_maxima))
    indices_faltantes = sorted(indices_totales - indices_presentes)

    print(f"\n🔎 Valores faltantes en la lista '{key}':")

    for idx in indices_faltantes:
        nombre = datos_concursos["Nombre"][idx] if idx < len(datos_concursos["Nombre"]) else "(sin nombre)"
        fecha_inicio = datos_concursos["Inicio"][idx] if idx < len(datos_concursos["Inicio"]) else "(sin fecha)"
        fecha_final = datos_concursos["Final"][idx] if idx < len(datos_concursos["Final"]) else "(sin fecha)"
        print(f"- Índice {idx} → concurso: {nombre} | Inicio: {fecha_inicio} | Final: {fecha_final}")




🔎 Valores faltantes en la lista 'Inicio':
- Índice 148 → concurso: CSI1* Opglabbeek | Inicio: (sin fecha) | Final: (sin fecha)

🔎 Valores faltantes en la lista 'Final':
- Índice 148 → concurso: CSI1* Opglabbeek | Inicio: (sin fecha) | Final: (sin fecha)

🔎 Valores faltantes en la lista 'Ambito':
- Índice 148 → concurso: CSI1* Opglabbeek | Inicio: (sin fecha) | Final: (sin fecha)


In [4]:
with open('../data/data_salto/pruebas/pruebas_salto_internacional_2025.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Disciplina es: 1144
La longitud de Fecha es: 1144
La longitud de Prueba es: 1144
La longitud de Categoría es: 1142
La longitud de Número es: 1141
La longitud de Concurso es: 1144


In [7]:
import json

# Cargar datos de pruebas
with open('../data/data_salto/pruebas/pruebas_salto_internacional_2020.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Longitud total según el campo más largo
longitud_total = len(datos_pruebas["Concurso"])

# Campos a verificar
campos_objetivo = ["Prueba", "Categoría", "Fecha"]  # puedes añadir más si quieres

# Verificar faltantes y mostrar con fecha si existe
for campo in campos_objetivo:
    if campo not in datos_pruebas:
        continue
    
    lista = datos_pruebas[campo]
    indices_faltantes = set(range(longitud_total)) - set(range(len(lista)))
    
    if indices_faltantes:
        print(f"\n🔍 Faltantes en '{campo}':")
        for i in sorted(indices_faltantes):
            fecha = datos_pruebas["Fecha"][i] if i < len(datos_pruebas["Fecha"]) else "(sin fecha)"
            concurso = datos_pruebas["Concurso"][i] if i < len(datos_pruebas["Concurso"]) else "(sin concurso)"
            print(f"- Índice {i} → concurso: '{concurso}' | fecha: {fecha}")





🔍 Faltantes en 'Prueba':
- Índice 14957 → concurso: 'CSI2* LIER (BEL)' | fecha: 115-125 - Two phases, both atc T A
- Índice 14958 → concurso: 'CSI2* LIER (BEL)' | fecha: CSI YH*
- Índice 14959 → concurso: 'CSI2* LIER (BEL)' | fecha: Disponibles
- Índice 14960 → concurso: 'CSI2* LIER (BEL)' | fecha: Salto de Obstáculos
- Índice 14961 → concurso: 'CSI2* LIER (BEL)' | fecha: 16/12/2020
- Índice 14962 → concurso: 'CSI2* LIER (BEL)' | fecha: 16/12/2020
- Índice 14963 → concurso: 'CSI2* LIER (BEL)' | fecha: 17/12/2020
- Índice 14964 → concurso: 'CSI2* LIER (BEL)' | fecha: 17/12/2020
- Índice 14965 → concurso: 'CSIYH LIER (BEL)' | fecha: 18/12/2020
- Índice 14966 → concurso: 'CSIYH LIER (BEL)' | fecha: 18/12/2020
- Índice 14967 → concurso: 'CSIYH LIER (BEL)' | fecha: 19/12/2020
- Índice 14968 → concurso: 'CSIYH LIER (BEL)' | fecha: 19/12/2020
- Índice 14969 → concurso: 'CSIYH LIER (BEL)' | fecha: 20/12/2020
- Índice 14970 → concurso: 'CSI2* LIER (BEL)' | fecha: (sin fecha)
- Índice 14971 → c

In [9]:
with open('../data/data_salto/concursos/concursos_salto_internacional_2025.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Nombre es: 149
La longitud de Categoría es: 149
La longitud de Provincia es: 149
La longitud de Localidad es: 149
La longitud de Disciplina es: 149
La longitud de Federación es: 149
La longitud de Resultados es: 149
La longitud de País es: 149
La longitud de Inicio es: 148
La longitud de Final es: 148
La longitud de Ambito es: 148


In [10]:
with open('../data/data_salto/pruebas/pruebas_salto_internacional_2025.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Disciplina es: 1144
La longitud de Fecha es: 1144
La longitud de Prueba es: 1144
La longitud de Categoría es: 1142
La longitud de Número es: 1141
La longitud de Concurso es: 1144


# Completo

In [38]:
with open('../data/data_completo/pruebas/pruebas_completo_nacional_2020.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Disciplina es: 561
La longitud de Fecha es: 561
La longitud de Prueba es: 561
La longitud de Categoría es: 561
La longitud de Número es: 561
La longitud de Concurso es: 561


In [39]:
with open('../data/data_completo/pruebas/pruebas_completo_internacional_2020.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Disciplina es: 496
La longitud de Fecha es: 496
La longitud de Prueba es: 496
La longitud de Categoría es: 496
La longitud de Número es: 496
La longitud de Concurso es: 496


In [40]:
with open('../data/data_completo/concursos/concursos_completo_internacional_2020.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Nombre es: 480
La longitud de Categoría es: 480
La longitud de Provincia es: 480
La longitud de Localidad es: 480
La longitud de Disciplina es: 480
La longitud de Federación es: 480
La longitud de Resultados es: 480
La longitud de País es: 480
La longitud de Inicio es: 477
La longitud de Final es: 477
La longitud de Ambito es: 477


In [41]:
with open('../data/data_completo/concursos/concursos_completo_nacional_2020.json', 'r', encoding='utf-8') as file:
    datos_pruebas = json.load(file)

# Ver la longitud de cada lista dentro del JSON
for key, value in datos_pruebas.items():
    if isinstance(value, list):  # Verificamos si el valor es una lista
        print(f"La longitud de {key} es: {len(value)}")

La longitud de Nombre es: 223
La longitud de Categoría es: 223
La longitud de Provincia es: 223
La longitud de Localidad es: 223
La longitud de Disciplina es: 223
La longitud de Federación es: 223
La longitud de Resultados es: 223
La longitud de País es: 223
La longitud de Inicio es: 223
La longitud de Final es: 223
La longitud de Ambito es: 223
